Hello @PaulSec, LGTM after testing.
Regarding the case, where a user wants to only detect the non presence of a header, without a specific value, he will have to set "header_name:". The colon seems not really intuitive, we should add that to the documentation to make it clear.

If the user provides the header name without colon :
``` bash
$ ./gochopchop scan -u https://github.com -c chopchop.yml
No vulnerabilities found.
panic: runtime error: index out of range [1] with length 1

goroutine 6 [running]:
gochopchop/pkg.ResponseAnalysis(0xc0000d6cf0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00002c690, 0x23, ...)
	/Users/tsoubiran/Documents/tools/ChopChop/pkg/resp_analysis.go:91 +0x7c2
gochopchop/app.Scan.func1(0xc000269e30, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000272860, 0xc0002b3090, 0x1, 0x1, ...)
	/Users/tsoubiran/Documents/tools/ChopChop/app/scan.go:144 +0x5f4
created by gochopchop/app.Scan
	/Users/tsoubiran/Documents/tools/ChopChop/app/scan.go:104 +0x9b8
```

With the following config : 

``` yaml
insecure: false
plugins:
  - uri: "/"
    checks:
      - name: strict transport security detection
        remediation: No Strict transport security has been declared, add one!
        description: Triggers if Strict Transport Policy has not been properly declared
        severity: "Low"
        no_headers:
            - "Strict-Transport-Security"
```

Thanks for this PR.

--- a/pkg/resp_analysis.go
+++ b/pkg/resp_analysis.go
@@ -86,6 +86,8 @@ func ResponseAnalysis(resp *http.Response, signature data.Check) bool {
 			// if the header has not been found, hit!
 			if !kFound {
 				return true
+			} else if kFound && len(pNoHeaders) == 1 { // if the header has not been specified.
+				return false
 			} else {
 				for _, n := range v { // usually, only one iteration
 					if strings.Contains(n, pNoHeaders[1]) {

